### PR TITLE
Create repo and export component

### PR DESCRIPTION
**Pull request by cmarin001**
This pull request announces the creation of a new repository for the `<Datefield />` component and its subsequent exportation for use across our application. Aimed at centralizing the development and maintenance of this critical date input component, the repository will serve as the primary source for the latest versions, ensuring consistent functionality and design integration within our digital ecosystem.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +34,13 @@
     "release": "git push --follow-tags && npm run auto release"
   },
   "dependencies": {
+    "@inubekit/foundations": "^2.5.0",
+    "@inubekit/icon": "^1.2.0",
+    "@inubekit/label": "^2.1.0",
+    "@inubekit/text": "^2.1.0",
     "react": "^18.2.0",
```
Comment (JanuarAsprilla) [inline package.json:41]: should be in peerDependences

Comment (cmarin001) [reply package.json:43]: solved

**Review by JanuarAsprilla — APPROVED** [review]

**Review by wfercanas — CHANGES_REQUESTED** [review]

File: src/Datefield/index.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        {required && !disabled && (
+          <Text
+            type="body"
+            size="small"
+            appearance="dark"
+            margin="0px 0px 0px 4px"
+            textAlign={"center"}
+          >
+            (Requerido)
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:96]: solved

File: vite.config.ts [hunk 3]
```diff
@@ -11,6 +11,23 @@ export default defineConfig({
       formats: ["es"],
       fileName: (format) => `index.${format}.js`,
     },
+    rollupOptions: {
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply vite.config.ts:14]: solved

File: src/Datefield/index.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+          $status={status}
+          $fullwidth={fullwidth}
+          $focused={focused}
+          onChange={onChange}
+          onFocus={interceptFocus}
+          onBlur={interceptBlur}
+        />
+      </StyledInputContainer>
+
+      {status === "invalid" && !disabled && (
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:124]: solved

File: src/Datefield/index.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+          />
+          <Text
+            type="body"
+            size="small"
+            textAlign="start"
+            margin="8px 0px 0px 4px"
+            appearance="danger"
+            disabled={disabled}
+          >
+            {message && `${message}`}
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:139]: solved

File: src/Datefield/index.tsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    onFocus,
+    onBlur,
+  } = props;
+
+  const [focused, setFocused] = useState(false);
+
+  const interceptFocus = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFocused(true);
+    if (typeof onFocus === "function") {
+      onFocus(e);
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:56]: solved

File: src/Datefield/styles.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
+import styled from "styled-components";
+
+import { inube } from "@inubekit/foundations";
+
+const $setBackgroundColor = ({ $readOnly }) =>
+  $readOnly && inube.label.content.color.regular;
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/styles.js:6]: solved

File: src/Datefield/styles.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  display: grid;
+  align-items: center;
+  box-sizing: border-box;
+  border-radius: 8px;
+  user-select: none;
+  padding-left: 16px;
+  padding-right: 16px;
+  pointer-events: ${({ $disabled }) => $disabled && "none"};
+  opacity: ${({ $disabled }) => $disabled && "0.5"};
+  background-color: ${$setBackgroundColor};
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/styles.js:30]: solved

File: src/Datefield/styles.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+  grid-template-columns: ${({ $iconBefore, $iconAfter }) => {
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/styles.js:31]: solved

File: src/Datefield/styles.js
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/styles.js:1]: solved

File: src/Datefield/index.tsx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+          $fullwidth={fullwidth}
+          $focused={focused}
+          onChange={onChange}
+          onFocus={interceptFocus}
+          onBlur={interceptBlur}
+        />
+      </StyledInputContainer>
+
+      {status === "invalid" && !disabled && (
+        <StyledMessageContainer $disabled={disabled} $status={status}>
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:125]: solved

File: src/Datefield/index.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+          </Text>
+        )}
+      </StyledContainerLabel>
+
+      <StyledInputContainer
+        $disabled={disabled}
+        $focused={focused}
+        $status={status}
+      >
+        <StyledInput
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:106]: solved

File: src/Datefield/index.tsx [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+  const interceptBlur = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFocused(false);
+    if (typeof onBlur === "function") {
+      onBlur(e);
+    }
+  };
+
+  return (
+    <StyledContainer $fullwidth={fullwidth} $disabled={disabled} $size={size}>
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/index.tsx:92]: solved

**Review by JanuarAsprilla — APPROVED** [review]

File: src/Datefield/styles.js [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+const StyledInputContainer = styled.div`
+  display: grid;
+  align-items: center;
+  box-sizing: border-box;
+  border-radius: 8px;
+  user-select: none;
+  padding: 0;
+  pointer-events: ${({ $disabled }) => $disabled && "none"};
+  opacity: ${({ $disabled }) => $disabled && "0.5"};
+  background-color:;
```
Comment (wfercanas) [inline src/Datefield/styles.js:25]: This property is empty.

File: src/Datefield/styles.js
Comment (wfercanas) [inline src/Datefield/styles.js:1]: Component breaks when fullwidth is true:
image

**Review by wfercanas — CHANGES_REQUESTED** [review]

File: src/Datefield/styles.js (hunk 13 above)
Comment (cmarin001) [reply src/Datefield/styles.js:25]: solved

File: src/Datefield/styles.js
Comment (cmarin001) [reply src/Datefield/styles.js:1]: solved

**Review by wfercanas — CHANGES_REQUESTED** [review]

File: src/Datefield/styles.js [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+`;
+
+const StyledMessageContainer = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: 16px;
+  pointer-events: none;
+  color: ${({ theme }) =>
+    theme?.input?.message?.color?.regular ||
+    inube.input.message.color.regular}};
```
[the remark this answers is not in the document]
Comment (cmarin001) [reply src/Datefield/styles.js]: solved

**Review by wfercanas — APPROVED** [review]

**Pull request merged**